src/server/svr_jobfunc.c

```diff
@@ -1526,83 +1639,63 @@ check_block(job *pjob, char *message)
-
-		sprintf(log_buffer,
-			"Unable to authenticate with , %s (%s:%d)", phost,
-			inet_ntoa(remote.sin_addr), remote.sin_port);
-
-		log_joberr(-1, __func__, log_buffer, jobid);
-		goto done;
+	blockj->ji_msg = malloc(strlen(message) + 1);
+	snprintf(blockj->ji_msg, strlen(message) + 1, "%s", message);
+	blockj->ji_submit_host = malloc(strlen(phost) + 1);
+	snprintf(blockj->ji_submit_host, strlen(phost) + 1, "%s", phost);
+	blockj->ji_port = port;
+	blockj->ji_reply_retries = 0;
+	blockj->ji_exitstat = pjob->ji_qs.ji_un.ji_exect.ji_exitstat;
+	snprintf(blockj->ji_jobid, sizeof(blockj->ji_jobid), "%s", pjob->ji_qs.ji_jobid);
+	
+	if ((blockj->ji_block_fd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+		sprintf(log_buffer, "%s: socket %s", __func__, strerror(errno));
+		log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_JOB, LOG_NOTICE,
+			jobid, log_buffer);
+		return;
 	}
 
-	/*
-	 **	All ready to talk... now send the info.
-	 */
-
-	DIS_tcp_setup(sock);
-	ret = diswsi(sock, 1);		/* version */
-	if (ret != DIS_SUCCESS)
-		goto err;
-	ret = diswst(sock, jobid);
-	if (ret != DIS_SUCCESS)
-		goto err;
-	ret = diswst(sock, message);
-	if (ret != DIS_SUCCESS)
-		goto err;
-	ret = diswsi(sock, pjob->ji_qs.ji_un.ji_exect.ji_exitstat);
-	if (ret != DIS_SUCCESS)
-		goto err;
-	(void)DIS_tcp_wflush(sock);
-
-	goto done;
-
-err:
-	sprintf(log_buffer, "%s: write %s(%s:%d) %s", __func__, phost,
-		inet_ntoa(remote.sin_addr), port, dis_emsg[ret]);
-	log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_JOB, LOG_NOTICE,
-		jobid, log_buffer);
-
-done:
-	if ((ret = CS_close_socket(sock)) != CS_SUCCESS) {
-
-		sprintf(log_buffer, "Problem closing security context, %s (%s:%d)",
-			phost, inet_ntoa(remote.sin_addr), port);
-
-		log_joberr(-1, __func__, log_buffer, jobid);
+	sock_flags = fcntl(blockj->ji_block_fd, F_GETFL, 0);
+	if (fcntl(blockj->ji_block_fd, F_SETFL, sock_flags | O_NONBLOCK) == -1) {
+		sprintf(log_buffer, "Unable to set non blocking flag for job %s", jobid);
+		log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_JOB, LOG_NOTICE, jobid, log_buffer);
+		close(blockj->ji_block_fd);
+		return;
 	}
 
-#ifdef WIN32
-	closesocket(sock);
-#else
-	close(sock);
-#endif
-
+	is_connect = connect(blockj->ji_block_fd, (struct sockaddr *)&remote, sizeof(remote));
+	connect_err = errno;
+	if (is_connect == -1) {
+		if (connect_err == EINPROGRESS || connect_err == ECONNREFUSED || connect_err == EADDRINUSE || (connect_err == ETIMEDOUT) ||
+			connect_err == ECONNREFUSED || connect_err == EWOULDBLOCK) {
+			ptask = set_task(WORK_Timed, time_now + 10, check_block_wt, blockj);
+			ptask->wt_parm2 = malloc(sizeof(int));
+			*(int *)ptask->wt_parm2 = 0;
+		}
+		else {
+			sprintf(log_buffer, "%s: Unable to connect to client %s for job %s", __func__, phost, jobid);
+			log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_JOB, LOG_NOTICE, jobid, log_buffer);
+			close(blockj->ji_block_fd);
+			return;
+		}
+	}
+	if (is_connect == 0) {
+		ptask = set_task(WORK_Timed, time_now + 10, check_block_wt, blockj);
```

Comment:
since this is connected right away, call the work_task function with WORK_immed to call it immediately